.circleci/config.yml

```diff
@@ -535,6 +540,8 @@ jobs:
     environment:
       EMCC_CORES: 16
       EMCC_USE_NINJA: 1
+      CCACHE_BASE_DIR: "ccache_dir"
+      EM_COMPILER_WRAPPER: "ccache"
```

Comment:
I don't know if this is enough.  I know that @juj had a custom patch for ccache and added a specific `_EMCC_CCACHE` feature to support it.  See #13681

Reply:
I think @dschuff is trying to integrate ccache in another way: in the backend between emcc and clang.

My emsdk support is placed in `user -> ccache -> emcc -> clang`. This looks like is doing `user -> emcc -> ccache -> clang`.

Would be fantastic to see what the performance difference of this approach ends up being.

Reply:
Performance-wise, the main difference would obviously be that we still have to run all the python code of the driver. Since this is a compile and not a link, there's not a huge amount of stuff that gets done, but certainly there's a little cost. Probably the builtin profiling support could estimate how much. But mostly I just picked this way because I didn't want to bother with a fork of ccache.

Reply:
My plan is to just key on the git version of clang or the emsdk installed by CI. Since I'm just trying to cache clang's output and not emscripten's, I think everything should be included in the clang version, the flags, the file inputs (i.e. the headers and sources).